File: app/src/main/java/com/app/id/receiver/NetworkBroadcastReceiver.java
package com.app.id.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkInfo;
import android.os.Build;

import com.app.id.application.Application;

public class NetworkBroadcastReceiver extends BroadcastReceiver {

    Application app;
    ConnectivityManager connectivityManager;

    public NetworkBroadcastReceiver(Application app, ConnectivityManager connectivityManager) {
        this.app = app;
        this.connectivityManager = connectivityManager;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        app.setNetworkConnected(isNetworkConnected(), isWifiConnected());
    }

    public boolean isNetworkConnected() {
        NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
        return activeNetwork != null && activeNetwork.isConnected();
    }

    @SuppressWarnings("deprecation")
    public boolean isWifiConnected() {
        if (Build.VERSION.SDK_INT < 21) {
            NetworkInfo wifiInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
            return wifiInfo.isConnectedOrConnecting();
        } else {
            Network[] networks = connectivityManager.getAllNetworks();

            for (Network network : networks) {
                NetworkInfo networkInfo = connectivityManager.getNetworkInfo(network);

                if (networkInfo == null) {
                    continue;
                }

                if (networkInfo.getType() != ConnectivityManager.TYPE_WIFI) {
                    continue;
                }

                if (!networkInfo.isConnectedOrConnecting()) {
                    continue;
                }

                return true;
            }

            return false;
        }
    }
}

File: app/src/main/java/com/app/id/receiver/NetworkBroadcastReceiver.kt
package com.app.id.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import android.os.Build
import com.app.id.application.Application

class NetworkBroadcastReceiver(var app: Application, var connectivityManager: ConnectivityManager) : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        app.setNetworkConnected(isNetworkConnected, isWifiConnected)
    }

    val isNetworkConnected: Boolean
        get() = connectivityManager.activeNetworkInfo?.let { it.isConnected } ?: false

    val isWifiConnected: Boolean
        get() {
            if (Build.VERSION.SDK_INT < 21) {
                @Suppress("DEPRECATION")
                val wifiInfo = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI)
                return wifiInfo.isConnectedOrConnecting
            } else {
                val networks = connectivityManager.allNetworks

                for (network in networks) {
                    val networkInfo = connectivityManager.getNetworkInfo(network) ?: continue

                    if (networkInfo.type != ConnectivityManager.TYPE_WIFI) {
                        continue
                    }

                    if (!networkInfo.isConnectedOrConnecting) {
                        continue
                    }

                    return true
                }

                return false
            }
        }
}
